src/test/java/com/kodilla/ecommercee/entity/ProductEntityTest.java

```diff
@@ -1,7 +1,113 @@
+        ProductEntity product = new ProductEntity("test name 1", new BigDecimal("1.11"), "test description 1");
+        product.setProductGroupEntity(groupEntity);
+        groupEntity.getProducts().add(product);
+        group = groupRepository.save(groupEntity);
+        productEntity = group.getProducts().get(0);
+        ProductEntity product2 = new ProductEntity("test name 2", new BigDecimal("2.22"), "test description 2");
+        productRepository.save(product2);
+    }
+
+    @After
+    public void cleanUpDatabase() {
+        groupRepository.deleteById(group.getId());
+    }
+
+    @Test
+    public void testCreateProductAndFind() {
+        //Given
+        ProductGroupEntity groupEntity = new ProductGroupEntity();
+        groupEntity.setGroupName("test group name 1");
+        Long groupId = groupRepository.save(groupEntity).getId();
+        ProductEntity productEntity = new ProductEntity("test name 1", new BigDecimal("1.11"), "test description 1");
+        productEntity.setProductGroupEntity(groupEntity);
+        groupEntity.getProducts().add(productEntity);
+        //When
+        Long id = productRepository.save(productEntity).getId();
+        Optional<ProductEntity> result = productRepository.findById(id);
+        //Then
+        assertTrue(result.isPresent());
+        assertEquals("test name 1", result.get().getName());
+        assertEquals(new BigDecimal("1.11"), result.get().getPrice());
+        assertEquals("test description 1", result.get().getDescription());
+        assertEquals(groupId, result.get().getProductGroupEntity().getId());
+        //CleanUp
+        groupRepository.deleteById(groupId);
+    }
+
+    @Test
+    public void testDeleteProductWithoutGroup() {
+        //Given
+        //When
+        group.getProducts().get(0).setProductGroupEntity(null);
+        Long productId = productRepository.save(group.getProducts().get(0)).getId();
+        group.setProducts(Collections.emptyList());
+        productRepository.deleteById(productId);
+        //Then
+        assertTrue(groupRepository.findById(group.getId()).isPresent());
+        assertFalse(productRepository.findById(productId).isPresent());
+    }
+
+    @Test
+    public void testUpdateProduct() {
+        //Given
+        //When
+        Optional<ProductEntity> result = productRepository.findById(productEntity.getId());
+        ProductEntity entity = result.get();
+        entity.setName("updated name");
+        entity.setPrice(new BigDecimal("2.22"));
+        entity.setDescription("updated description");
+        entity.setProductGroupEntity(null);
+        Long id = productRepository.save(entity).getId();
+        //Then
+        Optional<ProductEntity> updatedProduct = productRepository.findById(id);
+        assertTrue(updatedProduct.isPresent());
+        assertEquals(productEntity.getId(), updatedProduct.get().getId());
+        assertNotEquals(productEntity.getName(), updatedProduct.get().getName());
```

Comment:
Proponuję `assertEquals("updated name", updatedProduct.get().getName());`
Sprawdza to "więcej"